src/iuse.cpp

```diff
@@ -989,415 +990,279 @@ int iuse::chew(player *p, item *it, bool, const tripoint& )
-    // but best to be careful-KA101.
-    if( (it->type->can_use( "MYCUS" )) || (it->type->can_use( "MARLOSS" )) ||
-      (it->type->can_use( "MARLOSS_SEED" )) || (it->type->can_use( "MARLOSS_GEL" )) ) {
-        return 0;
-    }
-    if (p->has_trait( trait_THRESH_MARLOSS )) {
-        p->add_msg_player_or_npc( m_warning,
-            _("The %s sears your insides white-hot, and you collapse to the ground!"),
-            _("<npcname> writhes in agony and collapses to the ground!"),
-            it->tname().c_str());
-        p->vomit();
-        p->mod_pain(35);
-        // Lose a significant amount of HP, probably about 25-33%
-        p->hurtall(rng(20, 35), nullptr);
-        // Hope you were eating someplace safe.  Mycus v. Goo in your guts is no joke.
-        p->fall_asleep((3000 - p->int_cur * 10));
-        // Marloss is too thoroughly into your body to be dislodged by orals.
-        p->set_mutation( trait_MUTAGEN_AVOID );
-        p->add_msg_if_player(m_warning, _("That was some toxic %s!  Let's stick with Marloss next time, that's safe."), it->tname().c_str());
-        p->add_memorial_log(pgettext("memorial_male", "Suffered a toxic marloss/mutagen reaction."),
-                            pgettext("memorial_female", "Suffered a toxic marloss/mutagen reaction."));
-        return it->type->charges_to_use();
+    if( p.has_trait( trait_MUTAGEN_AVOID ) ) {
+         //~"Uh-uh" is a sound used for "nope", "no", etc.
+        p.add_msg_if_player( m_warning, _( "After what happened that last time?  uh-uh.  You're not drinking that chemical stuff." ) );
+        return mutagen_rejection::rejected;
     }
 
-    if (p->has_trait( trait_THRESH_MYCUS )) {
-        p->add_msg_if_player(m_info, _("This is a contaminant.  We reject it from the Mycus."));
-        if( p->has_trait( trait_M_SPORES ) || p->has_trait( trait_M_FERTILE ) ||
-            p->has_trait( trait_M_BLOSSOMS ) || p->has_trait( trait_M_BLOOM ) ) {
-            p->add_msg_if_player(m_good, _("We empty the %s and reflexively dispense spores onto the mess."));
-            g->m.ter_set(p->posx(), p->posy(), t_fungus);
-            p->add_memorial_log(pgettext("memorial_male", "Destroyed a harmful invader."),
+    static const std::vector<std::string> safe = {{
+        "MYCUS", "MARLOSS", "MARLOSS_SEED", "MARLOSS_GEL"
+    }};
+    if( std::any_of( safe.begin(), safe.end(), [it]( const std::string &flag ) {
+        return it.type->can_use( flag );
+    } ) ) {
+        return mutagen_rejection::accepted;
+    }
+
+    if( p.has_trait( trait_THRESH_MYCUS ) ) {
+        p.add_msg_if_player( m_info, _( "This is a contaminant.  We reject it from the Mycus." ) );
+        if( p.has_trait( trait_M_SPORES ) || p.has_trait( trait_M_FERTILE ) ||
+            p.has_trait( trait_M_BLOSSOMS ) || p.has_trait( trait_M_BLOOM ) ) {
+            p.add_msg_if_player( m_good, _( "We empty the %s and reflexively dispense spores onto the mess." ) );
```

Comment:
%s here isn't pointing to anything.

Reply:
It's an old error - it was like this before.
I'll change the message, the container is not accessible here and "we empty the mutagen" would not look good.